src/sentry/integrations/slack/threads/activity_notifications.py

```diff
@@ -15,3 +22,185 @@ class AssignedActivityNotification(BaseAssignedActivityNotification):
+                extra={
+                    "activity_id": self._activity.id,
+                },
+            )
+            provider = self.DEFAULT_PROVIDER_FALLBACK_TEXT
+
+        return provider.lower()
+
+    def get_ticket_number(self) -> str:
+        """
+        Returns the ticket number for the issue that was created on the external provider.
+        """
+        ticket_number = self._activity.data.get(self._TICKET_KEY, None)
+        if not ticket_number:
+            metrics.incr(
+                self._NO_LABEL_KEY_METRICS,
+                sample_rate=1.0,
+            )
+            _default_logger.info(
+                "Activity does not have a label key, using fallback",
+                extra={
+                    "activity_id": self._activity.id,
+                },
+            )
+            ticket_number = ""
+
+        return ticket_number
+
+
+class _AsanaExternalIssueCreatedActivity(_ExternalIssueCreatedActivity):
+    """
+    Override class for Asana as, at this time, the label, or ticket number, does not exist and has to be derived.
+    If plausible, this could be removed if the activity object itself properly has the correct data, but side effects
+    for that change are not yet known.
+    """
+
+    _DEFAULT_ASANA_LABEL_VALUE = "Asana Issue"
+
+    def get_ticket_number(self) -> str:
+        # Try to use the base logic if it works as a just in-case
+        stored_value = super().get_ticket_number()
+        if stored_value != "" and stored_value != self._DEFAULT_ASANA_LABEL_VALUE:
+            return stored_value
+
+        link = self.get_link()
+        if not link:
+            return ""
+
+        # Remove any trailing slashes
+        if link.endswith("/"):
+            link = link[:-1]
+
+        # Split the URL by "/"
+        parts = link.split("/")
+
+        # Get the last part
+        last_part = parts[-1]
+
+        return last_part
+
+
+def _external_issue_activity_factory(activity: Activity) -> _ExternalIssueCreatedActivity:
+    """
+    Returns the correct ExternalIssueCreatedActivity class based on the provider.
+    All classes have the same interface, the method for one is simply modified for its use case.
+    """
+    base_activity = _ExternalIssueCreatedActivity(activity=activity)
+    provider = base_activity.get_provider()
+    if provider == "asana":
+        return _AsanaExternalIssueCreatedActivity(activity=activity)
+
+    return base_activity
+
+
+class ExternalIssueCreatedActivityNotification(GroupActivityNotification):
+    metrics_key = "create_issue"
+    title = "External Issue Created"
+
+    def get_description(self) -> tuple[str, str | None, Mapping[str, Any]]:
+        external_issue = _external_issue_activity_factory(activity=self.activity)
+
+        provider = external_issue.get_provider()
+        # Use proper grammar, so use "an" if it's "external provider" and "a" if it's a regular name
+        if provider == external_issue.DEFAULT_PROVIDER_FALLBACK_TEXT:
```

Comment:
Asana should use "an" too?

Reply:
I guess so, could check for first letter being a vowel

Reply:
I'll pull that into the PR where I'll try to use the externalIssue model